update

Whisker17 opened the pull request:
Please have a look at this PR, looks like this is a big update for our documentation page

maltekliemann commented on docs/guide/create.md:
```diff
@@ -0,0 +1,42 @@
+   a market need to inject liquidity into their market, that is, at least 100
+   individual result tokens and 100 ZBS.
+
+:::tip
```
`:::` should be protected using `<!-- prettier-ignore -->` from prettier (unless you have a better solution)

maltekliemann commented on docs/learn/liquidity.md:
```diff
@@ -138,17 +138,17 @@ developed by Zeitgeist is based) leave plenty of opportunity for arbitrageurs.
 
 Video tutorials on liquidity pools:
 
--   [Finematics on Liquidity Pools](https://www.youtube.com/watch?v=cizLhxSKrAc)
```
When I run prettier on this file (and any other file), it inserts three whitespaces after the dash. Does prettier have a different effect for you?

Whisker17 replied:
When I run Prettier on my vscode, seems it won't insert 3 whitespaces, a little bit strange.

> When I run prettier on this file (and any other file), it inserts three whitespaces after the dash. Does prettier have a different effect for you?

maltekliemann replied:
Okay, riddle solved. The prettier integration for my editor uses prettier-1.0.0. My bad!

maltekliemann reviewed (changes requested):
Hey @Whisker17, that's indeed a big update! Good to see a guide focused on using the app in the docs 👍

- When I run `yarn build` or `yarn start`, I get the following error message:

  ```
  yarn run v1.22.17
  $ docusaurus build
  /Users/malte/documentation/node_modules/globby/index.js:3
  const arrayUnion = require('array-union');
                     ^
  
  Error [ERR_REQUIRE_ESM]: require() of ES Module /Users/malte/documentation/node_modules/array-union/index.js from /Users/malte/documentation/node_modules/globby/index.js not supported.
  Instead change the require of /Users/malte/documentation/node_modules/array-union/index.js in /Users/malte/documentation/node_modules/globby/index.js to a dynamic import() which is available in all CommonJS modules.
      at Object.<anonymous> (/Users/malte/documentation/node_modules/globby/index.js:3:20)
      at Object.<anonymous> (/Users/malte/documentation/node_modules/@docusaurus/utils/lib/globUtils.js:14:16)
      at Object.<anonymous> (/Users/malte/documentation/node_modules/@docusaurus/utils/lib/index.js:66:19)
      at Object.<anonymous> (/Users/malte/documentation/node_modules/@docusaurus/core/lib/server/index.js:11:17)
      at Object.<anonymous> (/Users/malte/documentation/node_modules/@docusaurus/core/lib/commands/build.js:17:18)
    at Object.<anonymous> (/Users/malte/documentation/node_modules/@docusaurus/core/lib/index.js:11:46)
    at async Promise.all (index 0) {
    code: 'ERR_REQUIRE_ESM'
  }
  error Command failed with exit code 1.
  info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
  ```

- I'm a bit confused about the folder `i18n`. Should that be comitted to the repository?
- Are you using prettier to format markdown files? Your formatter seems to have undone some of prettier's formatting.

sea212 commented:
I'll review once it builds the docs. I think we have to upgrade vercel's node version to >= 14 to be able to use the most recent docusaurus version.

Whisker17 commented:
> Thanks a lot for those huge improvements Whisker!
> 
> This review is only partial, still a lot is unreviewed. This is due to the following reasons:
> 
> 1. I am not sure what the `i18n` folder is for. Do we have to review that? Is this generated? Does it have to be committed to Github?
> 2. After applying some of the fixes specified in this review, the project still does not build. Please ensure that `yarn run build` completes successfully before requesting the next review.
> 3. Once ready for review, please also trigger Vercel manually to deploy the documentation if it does not do it automatically.

`i18n` is for our documentation localization by using Crowdin. We can use Github Action to upload and sync automatically from Crowdin. We don't need to change anything in `i18n` folder.

maltekliemann commented on docusaurus.config.js:
```diff
@@ -1,106 +1,150 @@
+          src: "img/Zeitgeist-trans.png",
+        },
+        items: [
+          {
+            to: "/", // getting-started',
+            label: "Docs",
+            position: "left",
+          },
+          {
+            type: "docSidebar",
+            sidebarId: "sdk",
+            label: "SDK",
+            position: "left",
+          },
+          {
+            type: "docSidebar",
+            sidebarId: "guide",
+            label: "App Guide",
+            position: "left",
+          },
```
I think we should delete these.

Generally, my plan for the wiki is to "steal" the style and structure of polkadot-wiki. This means that every chapter should be sorted into one of the "books", _General_, _Learn_, _Build_, _Maintain_. The SDK chapter and App chapters should not appear in the top bar, which must contain the books instead.

The polkadot-wiki has the books linked in the navbar. I don't think that's a good idea, since they are also available in the sidebar.

maltekliemann commented on docusaurus.config.js:
```diff
@@ -1,106 +1,150 @@
+              },
+            ],
+          },
+          {
+            href: "https://github.com/zeitgeistpm/documentation",
+            position: "right",
+            className: "header-github-link",
+            "aria-label": "GitHub repository",
+          },
```
Is this working properly for you? All this does on my machine in create a small whitespace between the localization menu and the search bar that links to our GitHub page. I think this can be deleted.

maltekliemann commented on docs/guide/redeem.md:
```diff
@@ -0,0 +1,23 @@
+# Redeem
+
+In the concept of the prediction market, the outcome token we purchased will
+eventually be settled into the original token, which is ZBS in our system, but
```
I think the App Guide should use ZTG instead of ZBS, even if that means that some of the screenshots are off.

maltekliemann commented:
I'm not a big fan of having to double-book all assets for every locale. But I'm not sure if there's a fix.

Whisker17 commented:
As for Crowdin action failure, the Crowdin team told me it's just because the PR hasn't been merged yet

maltekliemann commented:
That's weird, but I'm willing to take their word for it.

maltekliemann commented on crowdin.yml:
```diff
@@ -0,0 +1,12 @@
+project_id: 501094
+api_token: "CROWDIN_PERSONAL_TOKEN"
```
This is the way to go, I believe:

```suggestion
```

Whisker17 replied:
I think this can't help us fix the error. You can check the details [here](https://github.com/zeitgeistpm/documentation/runs/6362825406?check_suite_focus=true)

maltekliemann commented:
Have you added the secrets to this repository?

maltekliemann commented:
Oh, and another thing: If we automatically sync with crowdin, does this mean that we automatically commit anything that others submit to crowdin? I think that's a bad idea.

BTW, polkadot-wiki does not have crowdin actions - they seem to be doing this manually.

Whisker17 commented:
> Have you added the secrets to this repository?

Yep, you can check this screenshot.

![image](https://user-images.githubusercontent.com/12571049/167577919-bf692dd1-ae81-4886-965a-94978aa999ee.png)

Whisker17 commented:
> Oh, and another thing: If we automatically sync with crowdin, does this mean that we automatically commit anything that others submit to crowdin? I think that's a bad idea.
> 
> BTW, polkadot-wiki does not have crowdin actions - they seem to be doing this manually.

I think I've set the rule which the action will sync the Crowdin once a week, I'm not sure if it's a good idea, if not, I can delete thsi action.

maltekliemann commented:
So anyone could just upload some malicious content to crowdin, and if we don't check it in time, it gets committed? In that case, I suggest removing that workflow.

maltekliemann commented:
This might be the problem:
![167577919-bf692dd1-ae81-4886-965a-94978aa999ee](https://user-images.githubusercontent.com/49552287/167609808-eb09223c-07ae-47f2-8f86-bd18f1f9a802.png)

Whisker17 commented:
> This might be the problem: ![167577919-bf692dd1-ae81-4886-965a-94978aa999ee](https://user-images.githubusercontent.com/49552287/167609808-eb09223c-07ae-47f2-8f86-bd18f1f9a802.png)

But I think i've set SECRETs in the main repo.

maltekliemann commented:
Yeah, but a PR from a fork doesn't have access to the secrets (for good reason). You need to do the PR from a branch of this repository.

Whisker17 commented:
> 

So what do you think of this action, should I delete it?

maltekliemann commented:
Will we be able to pull crowdin submissions manually without the workflow?

Whisker17 commented:
> Will we be able to pull crowdin submissions manually without the workflow?

Yep, I can add an integration about this, and I will delete this workflow. Is that cool for you?

maltekliemann commented:
Yeah, that sounds cool :thumbsup:

Whisker17 commented:
I think it works now.

maltekliemann commented:
I'm pretty sure the config file is not correct this way. See https://support.crowdin.com/configuration-file/#api-credentials-from-environment-variables. By defining `project_id` and `api_token`, you are overriding `project_id_env` and `api_token_env`. Furthermore, `CROWDIN_PERSONAL_TOKEN` is most certainly not the API token, but rather the GitHub secret which contains the API token. And since that should remain a secret, we should not use `api_token`, but rather `api_token_env` for refering to the API token.

I'm also not sure if the project ID is supposed to be a secret. Do you know anything about that?

Whisker17 commented:
> I'm pretty sure the config file is not correct this way. See https://support.crowdin.com/configuration-file/#api-credentials-from-environment-variables. By defining `project_id` and `api_token`, you are overriding `project_id_env` and `api_token_env`. Furthermore, `CROWDIN_PERSONAL_TOKEN` is most certainly not the API token, but rather the GitHub secret which contains the API token. And since that should remain a secret, we should not use `api_token`, but rather `api_token_env` for refering to the API token.
> 
> I'm also not sure if the project ID is supposed to be a secret. Do you know anything about that?
> 
> Oh, also, you've removed the translation files from the config? Don't we need those?

Yep, I just deleted `api_token` and `project_id_env` and tested about that, it worked.

And I think the project ID can be showed to community, so I will delete the github secret about that.

As for the config I deleted, I was planning to have the `en` folder in the `i18n` folder as the source file, now I intend to directly use the `docs` as the source file, I will also delete the `en` folder in the `i18n`. What do you think of this?

Whisker17 commented:
The workflow for documentation will be:

1. Community translates on Crowdin.
2. Team members review the translations.
3. If there is no problems, we can sync those by Crowdin manually. It will push to my forked repo first (because they don't support Github Enterprise for free trial), and we can commit a PR then to merge to the main repo.

maltekliemann commented:
> As for the config I deleted, I was planning to have the `en` folder in the `i18n` folder as the source file, now I intend to directly use the `docs` as the source file, I will also delete the `en` folder in the `i18n`. What do you think of this?

Yeah, that sounds fine to me. :thumbsup:

> The workflow for documentation will be:
> 
> 1. Community translates on Crowdin.
> 2. Team members review the translations.
> 3. If there is no problems, we can sync those by Crowdin manually. It will push to my forked repo first (because they don't support Github Enterprise for free trial), and we can commit a PR then to merge to the main repo.

Seems fine, although the last sentence has me a bit confused. Can't we just get a subscription to crowdin that allows integration with our organization? Might use the opportunity and ramp up the number of allowed words.

Whisker17 commented:
> > As for the config I deleted, I was planning to have the `en` folder in the `i18n` folder as the source file, now I intend to directly use the `docs` as the source file, I will also delete the `en` folder in the `i18n`. What do you think of this?
> 
> Yeah, that sounds fine to me. 👍
> 
> > The workflow for documentation will be:
> > 
> > 1. Community translates on Crowdin.
> > 2. Team members review the translations.
> > 3. If there is no problems, we can sync those by Crowdin manually. It will push to my forked repo first (because they don't support Github Enterprise for free trial), and we can commit a PR then to merge to the main repo.
> 
> Seems fine, although the last sentence has me a bit confused. Can't we just get a subscription to crowdin that allows integration with our organization? Might use the opportunity and ramp up the number of allowed words.

Of course we can, I will double-check with our community manager about this.

Whisker17 commented:
@maltekliemann Just finished github integration and for now we can commit the translation to the main repo manually.

maltekliemann commented:
Cool, thanks Whisker! :thumbsup: We're almost done. We still need @saboonikhil to submit a review.

Also, I just realized that the search bar doesn't work. Is there something we can do about that?
<img width="1430" alt="Screenshot 2022-05-17 at 18 01 51" src="https://user-images.githubusercontent.com/49552287/168856875-233aa379-05b3-42c6-8596-5929717585e4.png">

saboonikhil commented:
@Whisker17 

The _Edit this page_ button on every page redirects to Github allowing changes to be pushed directly to `main` branch. Just wanted to understand if it is just available to me or everyone. I think we must have restrictions to the usage of this repository.

saboonikhil commented on docs/build/sdk/introduction.md:
```diff
@@ -0,0 +1,64 @@
+
+| Function Name                                                                           | Description                      |
+| --------------------------------------------------------------------------------------- | -------------------------------- |
+| [getBlockInfo](https://github.com/Whisker17/sdk-demo/tree/main/src/common#getblockinfo) | get block info in Zeitgeist.     |
```
So I am suggesting to replace `https://github.com/Whisker17/sdk-demo/tree/main/src/common#getblockinfo` with `/build/sdk/common#getblockinfo`. Just tested it!

Whisker17 commented:
> @Whisker17
> 
> The _Edit this page_ button on every page redirects to Github allowing changes to be pushed directly to `main` branch. Just wanted to understand if it is just available to me or everyone. I think we must have restrictions to the usage of this repository.

I've tested on Docusaurus page about this button, and it showed if you don't have the access to edit it, you need to commit a PR.

![image](https://user-images.githubusercontent.com/12571049/169727428-f5ecdbed-6614-49e1-b5d4-5ece7e6dfe49.png)

saboonikhil commented:
@Whisker17 I think there's some issue with resolving internal page urls, can be temporarily ignored using [onBrokenLinks](https://docusaurus.io/docs/api/docusaurus-config#onBrokenLinks) Docusaurus configuration.

Whisker17 commented:
> @Whisker17 I think there's some issue with resolving internal page urls, can be temporarily ignored using [onBrokenLinks](https://docusaurus.io/docs/api/docusaurus-config#onBrokenLinks) Docusaurus configuration.

Hey @saboonikhil thx for your advice. I've noticed this problem and had a test on my own local repo, everything went fine, so it's pretty strange and I've asked the docusaurus team about this, but we can try `onBrokenLinks` first.

Whisker17 commented:
BTW, @saboonikhil , do you like the structure for our sdk doc? (You can check [here](https://github.com/zeitgeistpm/documentation/pull/36/files#diff-772b4faa44e308a741f35b8d19e5e5df5b99feeab7699ae773522b0178496c4b)) If you have any idea about this, just lmk.

Whisker17 commented:
@saboonikhil fix the link probelm.

saboonikhil commented:
> BTW, @saboonikhil , do you like the structure for our sdk doc? (You can check [here](https://github.com/zeitgeistpm/documentation/pull/36/files#diff-772b4faa44e308a741f35b8d19e5e5df5b99feeab7699ae773522b0178496c4b)) If you have any idea about this, just lmk.

Yep, it's super cool tbh. Each function has a description, arguments table and an example. I am happy with the current state.

saboonikhil commented on docs/build/sdk/indexs.md:
````diff
@@ -0,0 +1,222 @@
+```
+
+[Code snippet](https://github.com/Whisker17/sdk-demo/tree/main/src/index/getAllMarkets.ts)
+
````
I think we need to add `createCpmmMarketAndDeployAssets` here as well.

Whisker17 replied:
> I think we need to add `createCpmmMarketAndDeployAssets` here as well.

You mean this?
![image](https://user-images.githubusercontent.com/12571049/169804200-465c94df-dc95-4a11-ac28-26f5ff0c75e5.png)

saboonikhil replied:
Yes, the url redirects to nowhere since `createCpmmMarketAndDeployAssets` isn't available at `Indexs` page.

Whisker17 replied:
> 

fixed it.

Whisker17 replied:
> Yes, the url redirects to nowhere since `createCpmmMarketAndDeployAssets` isn't available at `Indexs` page.

fixed it.

saboonikhil commented on docs/learn/guide/app.md:
```diff
@@ -0,0 +1,74 @@
+# APP Guide
```
Can same thing be done with the app guide? 

Instead of separate page on app guide, as soon as one clicks on app guide, they can have a walkthrough.

Whisker17 commented:
@maltekliemann I think I find why our search engine doesn't work. Since we don't have a landing page so the root url for search engine is a problem, and I've set the root url as `/docs` now, but we need to set up a redirect on our deploying. (So  that's why when you open [this](https://docs-n6rmajae3-zeitgeistpm.vercel.app/), it will show `page not find`, we need to set `example.com/docs` as default instead of `example.com`)

And I think our search engine works now.

maltekliemann commented:
@Whisker17 Cool, thanks! :thumbsup: Can you do the redirect?

maltekliemann commented:
Ok, great! A tiny complaint: Maybe replace everything in the blue box with the only a button saying "Zeitgeist Documentation"? The "1min" time marker seems a bit off.
